packages/flutter_tools/templates/app/lib/main.dart.tmpl

```diff
@@ -38,111 +38,182 @@ void main() {
   const MyApp({super.key});
 
-  // This widget is the root of your application.
   @override
   Widget build(BuildContext context) {
     return MaterialApp(
-      title: 'Flutter Demo',
+      title: 'Counter Example',
       theme: ThemeData(
-        // This is the theme of your application.
+        // This is the theme of the application.
+        //
+        // TRY THIS: Try running the application with "flutter run" on the
+        // command line, or the run button in the IDE. You'll see the
+        // application has a purple toolbar. Then, without quitting the app, try
+        // changing the colorScheme below to use a seedColor of Colors.green and
+        // then invoke "hot reload" (press "r" in the console where you ran
+        // "flutter run", save the changes to "hot reload" in a Flutter IDE, or
+        // press the "hot reload" button in a Flutter IDE).
         //
-        // Try running your application with "flutter run". You'll see the
-        // application has a blue toolbar. Then, without quitting the app, try
-        // changing the primarySwatch below to Colors.green and then invoke
-        // "hot reload" (press "r" in the console where you ran "flutter run",
-        // or simply save your changes to "hot reload" in a Flutter IDE).
         // Notice that the counter didn't reset back to zero; the application
-        // is not restarted.
-        primarySwatch: Colors.blue,
+        // state is not lost during the reload. This is what is meant by
+        // "stateful hot reload". To reset the state, use hot restart instead.
+        //
+        // This works for code too, not just values: Most code changes can be
+        // tested with just a hot reload.
+        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
+
+        // This makes the theme conform to a more up-to-date version of the
+        // Material Design specification. Soon, this will be the default, and
+        // this flag won't be necessary.
+        //
+        // See https://m3.material.io/ for more information about Material
+        // Design.
+        useMaterial3: true,
       ),
-      home: const MyHomePage(title: 'Flutter Demo Home Page'),
+      home: const MyHomePage(title: 'Counter Example'),
     );
   }
 }
 
+// This MyHomePage widget is the "home page" of the application. It is stateful,
+// meaning that it has a State object, defined below, that contains mutable
+// (modifiable) fields that affect how it looks and behaves.
+//
+// Widgets configure State objects. This object holds the immutable (unchanging)
+// values, in this case the title that is provided by the parent, which the
+// State object will use to configure its widgets.
+//
+// Fields in a Widget subclass are always marked "final" because Widget
+// subclasses are meant to be lightweight configuration objects that are rebuilt
+// with new values rather than modified. Modifiable state belongs in the
+// associated State object.
```

Comment:
I would probably lean towards using the second paragraph from the old version instead of these two new paragraphs. I agree that the new text is slightly more strictly correct, but I fear the nuance will be lost on the reader, and the additional text just risks that they'll not read any of it.